tests/test_args_near_unmapped.c

```diff
@@ -0,0 +1,78 @@
+#include <limits.h>
+
+#include "manager.h"
+
+/* Test wrapper for compartments with arguments.
+ *
+ * Takes at least two arguments: one argument for the compartment binary name,
+ * and one argument minimum to be passed to the compartment itself.
+ * TODO currently hard coding the entry points and which entry point to call,
+ * but plan to move this to some configuration file in the near future
+ */
+
+extern struct Compartment* loaded_comp;
+
+char*
+prep_filename(char* filename)
+{
+    const char* prefix = "./";
+    if (!strncmp(filename, prefix, strlen(prefix)))
+    {
+        filename += strlen(prefix);
```

Comment:
I'm pretty sure `+=` isn't doing the right thing here :)

Reply:
It's unused in this file (removed in https://github.com/capablevms/CHERI-ELF-comp/pull/7/commits/224dddc26070e8d66a5c361d016c49e01b2f4b14), but it is used in `manager_arg_passer.c` [1], where it seems to do what I want (i.e., remove the leading `./`):

```
(gdb) p filename
$5 = 0xffffffffee0b "./args_simple"
(gdb) f
#0  prep_filename (filename=0xffffffffee0b "./args_simple") at ../tests/manager_arg_passer.c:21
21              filename += strlen(prefix);
(gdb) n
22          }
(gdb) p filename
$6 = 0xffffffffee0d "args_simple"
```

[1] https://github.com/capablevms/CHERI-ELF-comp/pull/7/files#diff-6e0cbcab5be8419f8d8accdd49cf5e8096fefb3f0f6f3105b8cb95662ba16347R34

Reply:
Ah, I see, I misread the code -- ignore me!